src/luv.c

```diff
@@ -508,6 +508,56 @@ static void luv_handle_init(lua_State* L) {
   lua_setfield(L, LUA_REGISTRYINDEX, "uv_stream");
 }
 
+// Call lua function, will pop nargs values from top of vm stack and push some
+// values according to nresults. When error occurs, it will print error message
+// to stderr, and memory allocation error will cause exit.
+LUALIB_API int luv_cfpcall(lua_State* L, int nargs, int nresult, int flags) {
+  int ret, top, errfunc;
+
+  // Get the traceback function in case of error
+  if ((flags & LUVF_CALLBACK_NOTRACEBACK) == 0)
+  {
+    lua_pushcfunction(L, luv_traceback);
+    errfunc = lua_gettop(L);
+    // And insert it before the function and args
+    lua_insert(L, -2 - nargs);
+    errfunc -= (nargs+1);
+  }else
+    errfunc = 0;
+  top  = lua_gettop(L);
+
+  ret = lua_pcall(L, nargs, nresult, errfunc);
+  switch (ret) {
+  case LUA_OK:
+    break;
+  case LUA_ERRMEM:
+    fprintf(stderr, "System Error: %s\n", lua_tostring(L, -1));
+    if ((flags & LUVF_CALLBACK_NOEXIT) == 0)
+      exit(-1);
+    lua_pop(L, 1);
+    ret = -ret;
+    break;
+  case LUA_ERRRUN:
+  case LUA_ERRSYNTAX:
+  case LUA_ERRERR:
+  default:
+    fprintf(stderr, "Uncaught Error: %s\n", lua_tostring(L, -1));
```

Comment:
We could create another flag or reuse the `LUVF_CALLBACK_NOTRACEBACK` flag:

```
if ((flags & LUVF_CALLBACK_NOTRACEBACK) == 0) {
 fprintf(stderr, "Uncaught Error: %s\n", lua_tostring(L, -1));
}
```

Reply:
OK I think need a new flag LUVF_CALLBACK_NOERRMSG

Reply:
LUVF_CALLBACK_NOERRMSG will make callback don't traceback and not output any error message, but LUVF_CALLBACK_NOTRACEBACK only don't traceback to obtain more error message, but output origin message to stderr.